File: Java/src/common/IdlTestService.java
package common;

import ToyORB.ToyORBService;

@SuppressWarnings({"MethodParameterNamingConvention", "InstanceMethodNamingConvention", "InterfaceNamingConvention"})
public interface IdlTestService extends ToyORBService
{
    int getCounter();
    void updateCounter(int addition);
    String getImplementationName();
}

File: .NET Core/Servers/IdlTestService.cs
using Services;
using ToyORB;

namespace Servers
{
    public class IdlTestService : IIdlTestService
    {
        public string ServiceType { get; } = "IdlTestService";

        private int _counter = 0;
        public int getCounter()
        {
            return _counter;
        }

        public void updateCounter(int addition)
        {
            _counter += addition;
        }

        public string getImplementationName()
        {
            return "IDL Test service implemented in .NET Core (C#)";
        }
        public static void Main(string[] args)
        {
            IIdlTestService infoService = new IdlTestService();
            ToyNS.RegisterAndStart("IDL", infoService);
        }
    }
}
